player.py

```diff
@@ -1,12 +1,42 @@
+from dataclasses import dataclass
+from typing import List
+
+class PlayerState:
+    ALIVE : str = "alive"
+    DEAD: str = "dead"
+    BANNED: str = "banned"
+    all: List[str] = [ALIVE, DEAD, BANNED]
+
+# A player in mafia game
+@dataclass
 class Player:
-    """A player in mafia game"""
-    def __init__(self, ip, username, role, image_name):
-        self.ip = ip
-        self.username = username
-        self.state = "alive"
-        self.role = role
-        self.image_name = image_name
-        self.comment = False
+    ip : str
+    username : str
+    role : str
+    image_name : str
+    state : PlayerState = PlayerState.ALIVE
+    comment : bool = False
```

Comment:
Change it back.
No need for `typing`.

Reply:
data classes require you to do typing. (it just doesn't work if we don't)
needless to say, it is a much more intuitive way of making classes which are data holder.

Reply:
If you say it, that's OK.